grammars/grammars.json

```
@@ -589,6 +589,61 @@
 				"base": "python",
 				"file": "MagicRegExp.tmLanguage.json"
 			}
+		},
+		{
+			"scopeName": "inline-template.ng",
+			"grammar": {
+				"base": "angular",
+				"file": "inline-template.json"
+			},
+			"injectTo": ["source.ts"],
+			"embeddedLanguages": {
+				"text.html": "html",
+				"source.css": "css",
+				"source.js": "javascript"
+			}
+		},
+		{
+			"scopeName": "inline-styles.ng",
+			"grammar": {
+				"base": "angular",
+				"file": "inline-styles.json"
+			},
+			"injectTo": ["source.ts"],
+			"embeddedLanguages": {
+				"source.css": "css"
+			}
+		},
+		{
+			"scopeName": "template.ng",
+			"grammar": {
+				"base": "angular",
+				"file": "template.json"
+			},
+			"injectTo": ["text.html.derivative", "source.ts"],
+			"embeddedLanguages": {
+				"text.html": "html",
+				"source.css": "css"
+			}
+		},
+		{
+			"scopeName": "template.blocks.ng",
+			"grammar": {
+				"base": "angular",
+				"file": "template-blocks.json"
+			},
+			"injectTo": ["text.html.derivative", "source.ts"],
```

```suggestion
			"embeddedLanguages": {
				"text.html": "html",
				"source.css": "css"
			}
		},
		{
			"scopeName": "template.blocks.ng",
			"grammar": {
				"base": "angular",
				"file": "template-blocks.json"
			},
```

I definitely understand the concern about including these grammars even when the project doesn't use Angular. What about not injecting these into HTML and adding these grammars only for inline templates and styles? Those are much less aggressive in their matching and only match `styles`/`template` in a decorator. So you'd have a much lower risk of having Angular syntax highlighting show up in an unrelated project.

---

I think we’ll try including all grammars regardless of the project, and if that proves to be an issue for other HTML-based projects we may refine it.